rust/src/logs/conversion.rs

```diff
@@ -0,0 +1,159 @@
+            Some(v) => Some(v.into()),
+            None => None,
+        },
+        severity_text: match record.severity_text() {
+            Some(v) => Some(v.to_string()),
+            None => None,
+        },
+        severity_number: match record.severity_number() {
+            Some(v) => Some(v as u8),
+            None => None,
+        },
+        attributes: record.attributes_iter().map(|e| Some(e.into())).collect(),
+        trace_context: match record.trace_context() {
+            Some(v) => Some(v.into()),
+            None => None,
+        },
+        instrumentation_scope: Some(scope.into()),
+        resource: match resource {
+            Some(v) => Some(v.into()),
+            None => None,
+        },
+    }
+}
+
+impl From<&opentelemetry_sdk::logs::TraceContext> for TraceContext {
+    fn from(value: &opentelemetry_sdk::logs::TraceContext) -> Self {
+        Self {
+            trace_id: format!("{:x}", value.trace_id),
+            span_id: format!("{:x}", value.span_id),
+            trace_flags: match value.trace_flags {
+                Some(v) => Some(v.into()),
+                None => None,
+            },
+        }
+    }
+}
+
+impl From<&(opentelemetry::Key, opentelemetry::logs::AnyValue)> for KeyValue {
+    fn from(value: &(opentelemetry::Key, opentelemetry::logs::AnyValue)) -> Self {
+        Self {
+            key: value.0.to_string(),
+            value: value.1.clone().into(),
+        }
+    }
+}
+
+// The Rust SDK currently keeps the log `AnyValue` type separate from the general purpose `KeyValue`->`Value` type.
+// Per the OTel spec, they will likely be the same at some point.
+impl From<&opentelemetry::logs::AnyValue> for AnyValue {
+    fn from(value: &opentelemetry::logs::AnyValue) -> Self {
+        use opentelemetry::logs as otel;
+        match value {
+            otel::AnyValue::Boolean(v) => AnyValue::Value(AnyType::Boolean(v.clone())),
+            otel::AnyValue::Double(v) => AnyValue::Value(AnyType::Double(v.clone())),
+            otel::AnyValue::Int(v) => AnyValue::Value(AnyType::Integer(v.clone())),
+            otel::AnyValue::String(v) => AnyValue::Value(AnyType::String(v.to_string())),
+            otel::AnyValue::Bytes(v) => AnyValue::Value(AnyType::Bytes(v.to_vec())),
+            otel::AnyValue::ListAny(v) => AnyValue::List(
+                v.iter()
+                    .map(|e| match e {
+                        otel::AnyValue::Boolean(v) => AnyType::Boolean(v.clone()),
+                        otel::AnyValue::Double(v) => AnyType::Double(v.clone()),
+                        otel::AnyValue::Int(v) => AnyType::Integer(v.clone()),
+                        otel::AnyValue::String(v) => AnyType::String(v.to_string()),
+                        otel::AnyValue::Bytes(v) => AnyType::Bytes(v.to_vec()),
+                        // The `ListAny` and `Map` types are intentionally omitted because WIT doesn't support infinite recursion.
```

Comment:
Ooh. That's not good. What are we gonna do about that. That means we're not spec compliant.